### ISSUE TITLE
SNMP stats occasionally come back wrong, failing the tests

### ISSUE THREAD

**Issue by bossmc** [Metaswitch/clearwater-fv-test#22]
Repeatedly running the FV tests (with `while make run_test JUSTTEST="SNMPTest.*"; do date; done`) sometimes causes:

```
./fvtest /var/lib/jenkins/workspace/clearwater-fv-test/build/bin/fvtest --gtest_filter=*SNMPTest.** --gtest_output=xml:/var/lib/jenkins/workspace/clearwater-fv-test/build/testout/test_detail_fvtest.xml
Memcached port: 44444
Memcached PID:  20879
Running with random seed: 1439509493
Note: Google Test filter = *SNMPTest.**
[==========] Running 8 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 8 tests from SNMPTest
[ RUN      ] SNMPTest.ScalarValue
[       OK ] SNMPTest.ScalarValue (5 ms)
[ RUN      ] SNMPTest.TableOrdering
[       OK ] SNMPTest.TableOrdering (6 ms)
[ RUN      ] SNMPTest.LatencyCalculations
[       OK ] SNMPTest.LatencyCalculations (28 ms)
[ RUN      ] SNMPTest.CounterTimePeriods
[       OK ] SNMPTest.CounterTimePeriods (30 ms)
[ RUN      ] SNMPTest.IPCountTable
[       OK ] SNMPTest.IPCountTable (4 ms)
[ RUN      ] SNMPTest.SuccessFailCountTable
test_snmp.cpp:322: Failure
Value of: buf
  Actual: ".1.2.2.1.4.1 = Gauge32: 0\n"
Expected: ".1.2.2.1.4.1 = Gauge32: 1\n"
[  FAILED  ] SNMPTest.SuccessFailCountTable (29 ms)
[ RUN      ] SNMPTest.SingleCountByNodeTypeTable
test_snmp.cpp:340: Failure
Value of: buf
  Actual: ".1.2.2.1.2.1 = Gauge32: 0\n"
Expected: ".1.2.2.1.3.1.0 = Gauge32: 0\n"
[  FAILED  ] SNMPTest.SingleCountByNodeTypeTable (6 ms)
[ RUN      ] SNMPTest.SuccessFailCountByRequestTypeTable
test_snmp.cpp:406: Failure
Value of: buf
  Actual: ".1.2.2.1.3.1 = Gauge32: 0\n"
Expected: ".1.2.2.1.3.1.0 = Gauge32: 0\n"
[  FAILED  ] SNMPTest.SuccessFailCountByRequestTypeTable (5 ms)
[----------] 8 tests from SNMPTest (114 ms total)

[----------] Global test environment tear-down
[==========] 8 tests from 1 test case ran. (131 ms total)
[  PASSED  ] 5 tests.
[  FAILED  ] 3 tests, listed below:
[  FAILED  ] SNMPTest.SuccessFailCountTable
[  FAILED  ] SNMPTest.SingleCountByNodeTypeTable
[  FAILED  ] SNMPTest.SuccessFailCountByRequestTypeTable

 3 FAILED TESTS
make: *** [run_test] Error 1
```

This seems to occur about one time in 20, which suggests that the issue arises when a 5-second time window ticks over mid-test.

**Comment by rkd-msw** [comment]
Alan, you've already been looking at this a bit and you're happy to investigate further this sprint. Grab me if you need some advice.

**Comment by alango** [comment]
fixed by #23
